File: usedesk_sdk/src/main/java/ru/usedesk/sdk/internal/data/framework/api/standard/entity/request/SetEmailRequest.java
package ru.usedesk.sdk.internal.data.framework.api.standard.entity.request;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

public class SetEmailRequest extends BaseRequest {

    public static final String TYPE = "@@server/chat/SET_EMAIL";

    private final Payload payload;
    private final String email;

    public SetEmailRequest(@NonNull String token, @NonNull String email, @Nullable String name,
                           @Nullable Long phone, @Nullable Long additionalId) {
        super(TYPE, token);
        this.email = email;
        this.payload = new Payload(email, name, phone, additionalId);
    }

    class Payload {
        private final String email;
        private final String name;
        private final Long phone;
        private final Long additionalId;

        Payload(@NonNull String email, @Nullable String name, @Nullable Long phone, @Nullable Long additionalId) {
            this.email = email;
            this.name = name;
            this.phone = phone;
            this.additionalId = additionalId;
        }
    }
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/data/repository/api/loader/socket/_entity/setemail/SetEmailRequest.kt
package ru.usedesk.chat_sdk.data.repository.api.loader.socket._entity.setemail

import ru.usedesk.chat_sdk.data.repository.api.loader.socket._entity._extra.BaseRequest

internal class SetEmailRequest(
        token: String,
        email: String?,
        name: String?,
        note: String?,
        phone: Long?,
        additionalId: Long?
) : BaseRequest(TYPE, token) {

    private val payload: Payload = Payload(email, name, note, phone, additionalId)

    companion object {
        private const val TYPE = "@@server/chat/SET_EMAIL"
    }

    class Payload(
            private val email: String?,
            private val username: String?,
            private val note: String?,
            private val phone: Long?,
            private val additionalId: Long?
    )
}